File: TcpSocketSystem_Java/src/com/tecnm/sockets/usecases/procesareco/IProcesarEcoUseCase.java
package com.tecnm.sockets.usecases.procesareco;

import com.tecnm.sockets.domain.modelos.MensajeEco;

/**
 * Caso de Uso: Procesamiento y sanitización de mensajes de eco.
 */
public interface IProcesarEcoUseCase {

    MensajeEco ejecutar(String contenidoRaw, String emisor);
}

File: TcpSocketSystem_DotNet/src/UseCases/ProcesarEco/IProcesarEcoUseCase.cs
using TcpSocketSystem.Core.Domain.Modelos;

namespace TcpSocketSystem.Core.UseCases.ProcesarEco;

/// <summary>
/// Caso de Uso: Procesar Eco de Mensajes TCP.
/// Recibe una solicitud, valida/sanitiza el mensaje y construye la respuesta adecuada.
/// </summary>
public interface IProcesarEcoUseCase
{
    Task<MensajeEco> EjecutarAsync(string contenidoRaw, string emisor, CancellationToken cancellationToken = default);
}
